config/v1/types_dns.go

```diff
@@ -26,6 +26,34 @@ type DNSSpec struct {
 	// For example, given the base domain `openshift.example.com`, an API server
 	// DNS record may be created for `cluster-api.openshift.example.com`.
 	BaseDomain string `json:"baseDomain"`
+	// publicZone is the location where all the DNS records that are publicly accessible to
+	// the internet exist.
+	// If this field is nil, no public records should be created.
+	// +optional
+	PublicZone *DNSZone `json:"publicZone,omitempty"`
+	// privateZone is the location where all the DNS records that are only available internally
+	// to the cluster exist.
+	PrivateZone DNSZone `json:"privateZone"`
+}
+
+// DNSZone is used to define a DNS hosted zone.
+// A zone can be identified by an ID or tags.
+//
+// For example,
+// on AWS zone can be fetched using `ID` as [id][1] or using [resourcegroupstaggingapi][2] to fetch zone with filertags from `Tags`,
```

Comment:
nit: gofmt has [fairly minimal markup][1], so probably `[id][1]` -> `id [1]` and similar.  Run `godoc -html ./your/package` to check.

[1]: https://golang.org/pkg/go/doc/#ToHTML